### ISSUE TITLE
Transfer state machine should reset at the end of the saga

### ISSUE THREAD

**Issue by NicolasRouquette** [aklikic/lagom-fund-transfer-saga-scala#2]
The transfer state machine is defined in `com.klikix.fundtransfersaga.transfer.entity.transfer.TransferEntity`.

After a transaction is completed (successfully or rollbacked), the state machine remains in a non-empty state where it does not allow new transactions.

To allow new transactions, it seems that a couple of changes are needed:

- [ ] In `destinationAddStarted`, handling the `TransferSuccessful` event should result in the empty state.

- [ ] In `sourceRemoveRollbackStarted`, handling the `SourceRemoveRollbacked` event should result in the empty state.

It seems to me that if these corrections are made, then the behavior, as implemented, would correspond to the first option mentioned in the blog article:

> - only one transfer per source account – if one transfer is active, other are rejected
> - parallel transfers per source account
> - queuing of transfer requests per source account – if one transfer is active other transfer requests are not rejected but are stored in the LIFO queue

If that's the intent, it would be useful to clarify this explicitly somewhere.

**Comment by aklikic** [comment]
You are right. After transaction is finished entity state should move to initial state (notCreated).
It would be even more strength forward to rename state notCreated to notStarted. 

Regarding article options, this project implements option 1. I will point it out in the article. Tnx for suggestions.
It is trivial to implement option 2 by not using sourceAccountId for TransferEntity id but generate one for each transfer start.
Option 3 requires queueing transfer requests and polling next from queue when first is finished. Queue could be implemented using readside.

Nicolas would you like to send a PR for implementing move to initial state on finished transfer?

**Comment by NicolasRouquette** [comment]
I'd be glad to propose a PR as discussed above; however, I suggest reviewing this PR first:
https://github.com/aklikic/lagom-fund-transfer-saga-scala/pulls

Regarding names, I suggest a positive name like `idle` instead of a negative predicate like `notCreated` or `notStarted`. This could be applied to the state names in `AccountEntity` too.
